knox-ui/src/Main_components/KnowledgeGraph.js

```diff
@@ -0,0 +1,259 @@
+      // ==================== Add Link Names =====================
+      var linkTexts = svg.selectAll(".link-text")
+            .data(graph.triples)
+            .enter()
+            .append("text")
+              .attr("class", "link-text")
+              .text( function (d) { return d.p.label; })
+            ;
+  
+        //linkTexts.append("title")
+        //		.text(function(d) { return d.predicate; });
+      // ==================== Add Node =====================
+      var nodes = svg.selectAll(".node")
+                .data(filterNodesByType(graph.nodes, "node"))
+                .enter()
+                .append("circle")
+                  .attr("class", "node")
+                  .call(force.drag)
+              ;//nodes		
+      // ==================== Add Link Names =====================
+      var nodeTexts = svg.selectAll(".node-text")
+            .data(filterNodesByType(graph.nodes, "node"))
+            .enter()
+            .append("text")
+              .attr("class", "node-text")
+              .text( function (d) { return d.label; })
+            ;
+  
+        //nodeTexts.append("title")
+        //		.text(function(d) { return d.label; });
+      
+      
+      
+      // ==================== Force ====================
+      var color = d3.scale.category20();
+      force.on("tick", function() {
+        nodes
+          .attr("cx", function(d){ return d.x; })
+          .attr("cy", function(d){ return d.y; })
+          .attr("r", function(d){return d.weight+8})
+          .style("fill",function(d,i){return color(i);})
+          ;
+            
+        links
+          .attr("d", function(d) {
+              return "M" 	+ d.s.x + "," + d.s.y
+                    + "S" + d.p.x + "," + d.p.y
+                    + " " + d.o.x + "," + d.o.y;
+            })	
+          ;
+                   
+        nodeTexts
+          .attr("x", function(d) { return d.x + 12 ; })
+          .attr("y", function(d) { return d.y + 3; })
+          
+          
+          ;
+          
+  
+        linkTexts
+          .attr("x", function(d) { return 4 + (d.s.x + d.p.x + d.o.x)/3  ; })
+          .attr("y", function(d) { return 4 + (d.s.y + d.p.y + d.o.y)/3 ; })
+          
+          ;
+      });
+      
+      // ==================== Run ====================
+      force
+        .nodes(graph.nodes)
+        .links(graph.links)
+        .charge(-500)
+        .linkDistance(100)
+        .start()
+        ;
+        
+    }
+    function hej(){
+      var svg = d3.select("#svg-body").append("svg")
+				.attr("id",id())
+				.call(d3.behavior.zoom().on("zoom", function () {
+    			svg.attr("transform", "translate(" + d3.event.translate + ")   scale(" + d3.event.scale + ")")
+  				}))
+  				.append("g")
+				;
+				
+	var force = d3.layout.force().size([1000, 500]);
+	
+	var graph = triplesToGraph(svg);
+  console.log(graph);
+  update(svg, force, graph);
+  triples = [];
+    }
+    function id(){
+      return "kl";
+    }
```

Comment:
Der er ingen grund til at bruge 'function' hvis den alligevel altid returnere det samme.
```suggestion

    const id = "kl";
```